### PR TITLE
add v8::SharedArrayBuffer

### PR DESCRIPTION
**Pull request by afinch7**
It's technically possible to drop the `SharedBuf` returned by `SharedArrayBuffer::new`(and end up with some strange results). Maybe I should rework this to use the internal allocation/backing instead.

### CODE REVIEW

**Comment by ry** [comment]
@afinch7 we’ve recently added more support for BackingStore in ArrayBuffer. Can you make use of that?

**Comment by afinch7** [comment]
@ry I have a working version of this using the internal backing, but it might make sense to add support for external backing via `BackingStore` in a future PR.

File: src/binding.cc [hunk 1]
```diff
@@ -682,6 +682,26 @@ v8::PrimitiveArray* v8__ScriptOrModule__GetHostDefinedOptions(
...
+  return local_to_ptr(v8::SharedArrayBuffer::New(isolate, byte_length));
+}
+
+size_t v8__SharedArrayBuffer__ByteLength(v8::SharedArrayBuffer& self) {
+  return self.ByteLength();
+}
+
+void v8__SharedArrayBuffer__GetContents(v8::SharedArrayBuffer& self, v8::SharedArrayBuffer::Contents* out) {
+  *out = self.GetContents();
+}
```
Comment (ry) [inline src/binding.cc]: SharedArrayBuffer::GetContents is deprecated:
https://cs.chromium.org/chromium/src/v8/include/v8.h?l=5698&rcl=4c53f9a51444393133ff303952f1296603d44ab7

Comment (afinch7) [reply src/binding.cc]: Didn't see that will refactor to use `BackingStore`.

**Pull request merged**